File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);  // 오름차순 정렬

        int left = 0;
        int right = people.length - 1;
        int boatCount = 0;

        while (left <= right) {
            if (people[left] + people[right] <= limit) {
                // 가벼운 사람 + 무거운 사람 같이 탑승
                left++;
            }
            // 무거운 사람은 무조건 한 번 탐
            right--;
            boatCount++;
        }

        return boatCount;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    // 몸무게 오름차순 정렬
    people.sort((a, b) => a - b);

    let left = 0;
    let right = people.length - 1;
    let boatCount = 0;

    while (left <= right) {
        // 가장 가벼운 사람과 가장 무거운 사람이 함께 탈 수 있는 경우
        if (people[left] + people[right] <= limit) {
            left++;
        }

        // 가장 무거운 사람은 항상 탑승
        right--;
        boatCount++;
    }

    return boatCount;
}